src/librustc_infer/infer/mod.rs

```diff
@@ -1151,6 +1154,17 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         r
     }
 
+    pub fn probe_full<R, F>(&self, f: F) -> R
```

Comment:
This definitely needs a comment -- what is the role of `probe` vs `probe_full`? It seems that `probe_full` saves more information, but when would you want to do that?

Reply:
I can try to dig into why, but I just noticed that there was only a few `probe` calls that actually used the extra fields that are now in `FullSnapshot` all other calls to `start_snapshot` ignored them so it seemed a waste to record them.